### PR TITLE
Make test crypt be test function

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #1473

The PR has tested pass by checking local log(mbedtls/openssl)  and running on my branch.

For commit [Make test_crypt be standard test function](https://github.com/DMTF/libspdm/commit/1197308950cd81a9e5197714b77fdd391c780c0b) :

- Add some macro like LIBSPDM_RSA_SSA_SUPPORT_TEST;
- For some only supported crypt function in opesnssl, skip the test when mbedtls build.
- For some crypt function which both mbedtls and openssl don't support, skip them.
- Make all return right.

For commit [Enable test_crypt in CI again](https://github.com/DMTF/libspdm/commit/85eca38152b742ea65ecc3a2bf4271822af575cb):

- enable CI.

Signed-off-by: Wenxing Hou <wenxing.hou@intel.com>

### CODE REVIEW

**Review by steven-bellock — APPROVED** [review]

File: unit_test/test_crypt/aead_verify.c [hunk 1]
```diff
@@ -252,57 +252,8 @@ bool libspdm_validate_crypt_aead_cipher(void)
...
-    if (libspdm_const_compare_mem(OutBuffer, m_libspdm_sm4_gcm_pt,
-                                  sizeof(m_libspdm_sm4_gcm_pt)) != 0) {
-        libspdm_my_print("[Fail]");
-        goto Exit;
-    }
-
-    libspdm_my_print("[Pass]");
-Exit:
+    /* the sm4_gcm is not support now*/
+    libspdm_my_print("\n- SM4-GCM Encryption is not support");
```
Comment (jyao1) [inline unit_test/test_crypt/aead_verify.c:256]: why you **comment** unsupported sm2, but **delete** unsupported sm4 here?
We need use the consistent rule.

Maybe just comment the unsupported code.

File: unit_test/test_crypt/test_crypt.c [hunk 2]
```diff
@@ -150,7 +153,7 @@ int main(void)
 {
     int return_value = 0;
 
-    if (libspdm_cryptest_main() != 0) {
+    if (libspdm_cryptest_main() == false) {
```
Comment (jyao1) [inline unit_test/test_crypt/test_crypt.c:156]: Don't compare with bool (true/false).

just use `!libspdm_cryptest_main()`.

File: unit_test/test_crypt/CMakeLists.txt [hunk 3]
```diff
@@ -38,6 +38,10 @@ SET(test_crypt_LIBRARY
     spdm_crypt_ext_lib
 )
 
+if(CRYPTO STREQUAL "openssl")
+    add_definitions(-DLIBSPDM_USE_OPENSSL=1)
```
Comment (jyao1) [inline unit_test/test_crypt/CMakeLists.txt:42]: If `LIBSPDM_USE_OPENSSL` is false, there is no guarantee some crypto functions are enabled/disabled.

For consistency, I recommend:
1) we define a test_mask, for all functions, match - crypto configuration in spdm_lib_config.h
```
LIBSPDM_RSA_SSA_SUPPORT_TEST
LIBSPDM_RSA_PSS_SUPPORT_TEST
LIBSPDM_ECDSA_SUPPORT_TEST
LIBSPDM_SM2_DSA_SUPPORT_TEST
LIBSPDM_EDDSA_ED25519_SUPPORT_TEST
LIBSPDM_EDDSA_ED448_SUPPORT_TEST
LIBSPDM_FFDHE_SUPPORT_TEST
LIBSPDM_ECDHE_SUPPORT_TEST
LIBSPDM_SM2_KEY_EXCHANGE_SUPPORT_TEST
LIBSPDM_AEAD_GCM_SUPPORT_TEST
LIBSPDM_AEAD_CHACHA20_POLY1305_SUPPORT_TEST
LIBSPDM_AEAD_SM4_SUPPORT_TEST
LIBSPDM_SHA256_SUPPORT_TEST
LIBSPDM_SHA384_SUPPORT_TEST
LIBSPDM_SHA512_SUPPORT_TEST
LIBSPDM_SHA3_256_SUPPORT_TEST
LIBSPDM_SHA3_384_SUPPORT_TEST
LIBSPDM_SHA3_512_SUPPORT_TEST
LIBSPDM_SM3_256_SUPPORT_TEST
```

2) We may use `LIBSPDM_USE_OPENSSL` to configure above macro.
3) We may use `LIBSPDM_USE_MBEDTLS` to configure above macro as well.

File: unit_test/test_crypt/test_crypt.c [hunk 4]
```diff
@@ -120,10 +120,13 @@ bool libspdm_cryptest_main(void)
...
-    if (!status) {
-        return status;
-    }
+
+    /* the Crypto SM2 key Exchange is not support now
+     * status = libspdm_validate_crypt_sm2();
+     * if (!status) {
+     *     return status;
+     * }
+     **/
```
Comment (jyao1) [inline unit_test/test_crypt/test_crypt.c:129]: If we can add crypto configuration for test, then we dont need comment or delete.
We can just disable it by macro.

**Review by jyao1 — CHANGES_REQUESTED** [review]
Please use crypto test configuration to isolate unsupported algo test.

File: unit_test/test_crypt/test_crypt.c (hunk 2 above)
Comment (Wenxing-hou) [reply unit_test/test_crypt/test_crypt.c]: Thanks. I have changed it.

File: unit_test/test_crypt/CMakeLists.txt (hunk 3 above)
Comment (Wenxing-hou) [reply unit_test/test_crypt/CMakeLists.txt]: Thanks. I have added many macro in CMakelists.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**